sagemathinc/cocalc
# webapp: some enhancements for new filenames

# Description
This includes some changes to the dialog asking for a filename, i.e. the +new dropdown **without** entering any filename in the search box. It would be consequential to also add this to the larger "+new" dialog, but that's for later. 

# Testing Steps
1. dropdown select a filename, random filename appears
2. when you select python or sage, there are underlines in the name
3. there is an account setting for the type of random filename, which is also exposed in a dropdown.

# Relevant Issues

### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):

Front end:
